src/parser.rs

```diff
@@ -1440,22 +1511,28 @@ pub fn parse_struct(
                             });
                             last_visibility = None;
 
-                            let (mut var_decl, err) =
-                                parse_variable_declaration(tokens, index, visibility);
+                            let (parsed_var, err) = parse_variable(tokens, index, false);
                             error = error.or(err);
 
-                            // Ignore immutable flag for now
-                            var_decl.mutable = false;
-
-                            if var_decl.parsed_type == ParsedType::Empty {
+                            if parsed_var.parsed_type == ParsedType::Empty {
                                 trace!("ERROR: parameter missing type");
 
                                 error = error.or(Some(JaktError::ParserError(
                                     "parameter missing type".to_string(),
-                                    var_decl.span,
+                                    parsed_var.span,
                                 )))
                             }
 
+                            let var_decl = ParsedFieldDecl {
+                                name: parsed_var.name,
+                                parsed_type: parsed_var.parsed_type,
+                                mutable: false,
+                                span: parsed_var.span,
+                                visibility,
+                            };
+
+                            // Ignore immutable flag for now
```

Comment:
If this is meant to be something to fix in the future, please add a `FIXME:` so it's clear. Otherwise it's fine.

Reply:
I didn't add that. It was already there, but I should sure it is in the right place.